runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5235,11 +5239,18 @@ class INTERPRETER_CLASS
 				UDATA result = preparePinnedVirtualThreadForUnmount(_currentThread, object, true);
 				VMStructHasBeenUpdated(REGISTER_ARGS);
 				if (J9_OBJECT_MONITOR_OOM != result) {
+					PORT_ACCESS_FROM_JAVAVM(_vm);
 					/* Handle the virtual thread Object.wait call.
 					 * VirtualThread.timeout is a private field used by both VM and JCL to temporarily hold
 					 * the value of expected wait/park time before a wake up task is scheduled using the value.
 					 */
-					J9VMJAVALANGVIRTUALTHREAD_SET_TIMEOUT(_currentThread, _currentThread->threadObject, millis + (nanos / 1000000));
+					I_64 timeout = millis + (nanos / 1000000);
+					J9VMJAVALANGVIRTUALTHREAD_SET_TIMEOUT(_currentThread, _currentThread->threadObject, timeout);
```

Comment:
`VirtualThread` treats `timeout` as a number of nanoseconds - see https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/src/java.base/share/classes/java/lang/VirtualThread.java#L579.

Reply:
`timeout` is treated as nanoseconds for parking, it is treated as milliseconds for waiting. 
see https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/src/java.base/share/classes/java/lang/VirtualThread.java#L624

Reply:
The comment specifically mentions "park" in which case the value would be nanoseconds. Please correct the comment and expand it to explain why milliseconds are the right units here and on line 5812 (where another comment would be helpful).

Reply:
added comment for line 5812